docs/releasing/publishing-a-pre-release.md

```diff
@@ -1,19 +1,33 @@
-# Publishing pre-release of GOV.UK Frontend
+# Publish pre-release of GOV.UK Frontend
 
-We use pre-releases when:
-- We are working on a new contribution and want to see it in the Design System website
-- We want to try an experimental feature as if it's pushed to npm.
+Use pre-releases when you:
+- [work on developing a component or pattern](https://design-system.service.gov.uk/community/develop-a-component-or-pattern/) for the GOV.UK Design System
+- want to trial an experimental feature (guidance on trialling experimental features is in development)
 
-This is done by pushing the files used for a GOV.UK Frontend release (the contents of the `package` directory) to a new branch which can be installed by npm as if it was a released npm package.
+> :warning:Your projects should never depend on a pre-released GOV.UK Frontend package. This is because someone could remove the GitHub branch containing the pre-release package at any time. For this reason, never use a pre-released package in a production setting.
 
-No changes get published to npm as part of the process.
+## What happens when you pre-release GOV.UK Frontend
 
-1. Checkout the branch you want to pre-release and pull latest changes.
+When you pre-release GOV.UK Frontend, this creates a GitHub branch. This branch contains the GOV.UK Frontend `/package` directory with your trial changes.
 
-2. Run `nvm use` to ensure you are using the right version of Node.js and npm.
+Projects can point to this branch in their package.json, instead of to the published [GOV.UK Frontend npm package](https://www.npmjs.com/package/govuk-frontend). No changes are published to the GOV.UK Frontend npm package as part of this process.
 
-3. Run `npm install` to ensure you have the latest dependencies installed.
+## Publish a pre-release
 
-4. Run `npm run pre-release`. This will create and push a new branch `pre-release-[your-branch-name-here]` to GitHub.
+1. Run `git checkout -b BRANCH-FOR-PRE-RELEASE` to create a new local branch of the GOV.UK Frontend repo.
```

Comment:
The scope of this has changed from 'publishing a pre-release (of some work that's already been done)' to 'making changes to frontend and then publishing them as a pre-release'.

I can see the benefit of doing so, but I'm wondering if either the title needs to change or we need some sort of introductory text. I think there's a good chance that someone comes in to this document having already made the changes on a branch, and if they follow these instructions they're going to create a branch that they don't need and potentially get quite confused when they reach step 5.

Reply:
I don't think the current title (Publish pre-release of GOV.UK Frontend) is as descriptive as it could be. It tells users what they'll do, but not why they should do it.

Could 'Preview changes with pre-releases' work as a title? @36degrees @hannalaakso

Reply:
The more I think about it, the less sure I am about this change.

As it stands, I don't think step 5 provides enough detail for the user to actually be able to do it. Summarising it as 'make the changes' glosses over a huge amount of information that most users are going to need – for example, where to actually make the changes, how to know whether the changes have worked (for example how to run the review app) and how to deal with issues that come up (for example, the linter raising errors)

If we actually added that detail, the scope of this document would need to change entirely in order to accommodate it. Without the detail, it's incomplete and forces the user to break out of the steps and go and read other documentation.

With that in mind, could we consider scoping this tightly around the steps to publish a pre-release (assuming you have already made the changes you want to make) and to instead frame the guidance so that users understand how it fits into the wider context and exactly what we expect them to have already done?

Reply:
@36degrees Not sure I follow — can we have a quick call tomorrow to flesh out what that scoping would look like?

Reply:
Sure thing. It'd be good to invite @hannalaakso as well.

Reply:
Whoops, forgot to resolve this. Will do that now I've sent the updated content for review.

Reply:
Whoops, forgot to resolve this. Will do that now I've sent the updated content for review.